### PR TITLE
tRPC routing

### PR DESCRIPTION
**Pull request by theo-thanhlam**
The main routing is under src/routes/blog, where all the tRPC things happen. The attached video shows how FE can use it
[tRPC showcase](https://drive.google.com/file/d/13Rtunnl0Ta0FCEcAFTM1QefNAJg_aLdi/view?usp=sharing)

Some minor points I could do better but need advising:
_ How can I restructure the code in src/routes/blog.ts? it looks really messy. I try to create a separate file that only stores the requirement for the procedure (src/services/blog.ts) but it didn't work.
_ The limitation of the database for column description and content is 191, but BlogSchema has different limitation for that: 512 chars for description and 16384 chars for content. I don't want to directly adjust on  planetscale so what should I do ?

Please let me know how I did
Thanks

### CODE REVIEW

**Comment by samsoncn** [comment]
1. My suggestion is that you can first create all service functions in `src/services/blog.ts`, then recall the functions and do the trpc router in `src/routes/blog.ts:`

src/services/blog.ts:

> import {
>     createBlog as dbCreateBlog,
> } from "../lib/data";
> export async function getBlogsService() {
>     return await dbGetBlogs();
> }

src/routes/blog.ts:

> export const blogRouter = router({
>     getBlogs: publicProcedure.query(() => {
>         return BlogService.getBlogsService();
>     }),
>     createBlog: publicProcedure.input(BlogSchema).mutation((opts) => {
>         return BlogService.createBlogService(opts.input);
>     }),

2. we can just set it back to 191 characters max

**Comment by samsoncn** [comment]
we will need to add `server/package-lock.json` to .gitignore

**Review by samsoncn — APPROVED** [review]

**Review by samsoncn — APPROVED** [review]

File: server/src/app.ts
Comment (nikhilsharma26500) [inline server/src/app.ts:1]: change default port to something other than 3000 since it's frontend's default port

File: server/src/lib/data.ts
Comment (nikhilsharma26500) [inline server/src/lib/data.ts:1]: Ignoring this for now since it is to be looked into by database team

File: server/src/services/blog.ts
Comment (nikhilsharma26500) [inline server/src/services/blog.ts:1]: Add some documentation above functions to explain about their parameters and expected results.

**Review by nikhilsharma26500 — APPROVED** [review]

**Pull request merged**